go/proxyd/rpc.go

```diff
@@ -152,3 +164,52 @@ func IsBatch(raw []byte) bool {
 	}
 	return false
 }
+
+func isArchiveRequest(req *RPCReq, getBlockNum GetLatestBlockNumFn) (bool, error) {
+	const BLOCK_NUM_ARCHIVE_HEIGHT = 256
+
+	var (
+		tag            string
+		latestBlockNum uint64
+		err            error
+	)
+	switch req.Method {
+	case "eth_call", "eth_getBalance", "eth_getTransactionCount":
+		tag, err = extractTerminalBlockTag(req)
+	case "eth_getBlockByNumber":
+		tag, _, err = decodeGetBlockByNumberParams(req.Params)
+	default:
+		return false, nil
+	}
+
+	if err != nil {
+		return false, err
+	}
+	if tag == "" || isBlockDependentParam(tag) {
+		return false, nil
+	}
+	// NOTE: we're reading from an in-memory getBlockNum LVC, which doesn't block
+	if latestBlockNum, err = getBlockNum(context.Background()); err != nil {
+		return false, err
+	}
+
+	blockNum, err := decodeBlockInput(tag)
+	if err != nil {
+		return false, err
+	}
+	return blockNum+uint64(BLOCK_NUM_ARCHIVE_HEIGHT) < latestBlockNum, nil
+}
+
+// quick hack: if the request looks like a request containing a quantity|tag at the end, then return the tag
+func extractTerminalBlockTag(req *RPCReq) (string, error) {
+	var input []json.RawMessage
+	if err := json.Unmarshal(req.Params, &input); err != nil {
+		return "", err
+	}
+	if len(input) < 2 {
```

Comment:
are there any requests that take only a block tag?

Reply:
also note that that function is only [called for eth_call, eth_getBalance and eth_getTransactionCount rpcs ](https://github.com/ethereum-optimism/optimism/pull/2635/files#diff-326039202cfd8c96893c534b0a1be54828dbaf252544893173db3d3d85579448R177)

Reply:
all of those take more than 2 elements so we good